core/src/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -1,40 +1,250 @@
 use rustfsm::{fsm, TransitionResult};
+use temporal_sdk_core_protos::{
+    coresdk::workflow_commands::UpsertWorkflowSearchAttributes,
+    temporal::api::{
+        command::v1::Command,
+        enums::v1::{CommandType, EventType},
+        history::v1::HistoryEvent,
+    },
+};
+
+use crate::workflow::machines::{Cancellable, EventInfo, MachineKind, WFMachinesAdapter};
+
```

Comment:
Nit: Collapse these spaces everywhere and use `cargo +nightly fmt` (need to install nightly rust) which will apply the nesting formatting (annoyingly not stabilized yet.

The only place I use spaces between imports is when they are `pub use`d to differentiate re-exports.